doc/whats_new.rst

```diff
@@ -139,6 +139,8 @@ API
 
     - Added :class:`mne.decoding.UnsupervisedSpatialFilter` providing interface for scikit-learn decomposition algorithms to be used with MNE data, by `Jean-Remi King`_ and `Asish Panda`_
 
+    - class:`mne.decoding.FilterEstimator` has been deprecated in favor of :class:`mne.decoding.Filterer` by `Asish Panda`_
```

Comment:
It's a tiny change of API, we want to pass only explicit and necessary arguments (i.e. not info, but sfreq), and deal with numpy arrays. Deprecation within the same class would be a pain because the first necessary init argument to FilterEstimator is info.